### PR TITLE
Bugfix for multi salt-minion process startup

### PR DESCRIPTION
**Pull request by rongzeng54**
### What does this PR do?
Bugfix for multi salt-minion process startup

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/50507

### Previous Behavior
Starts slowly, does not exclude other startup action during startup.

### New Behavior
Starts slowly, does exclude other startup action during startup by lockfile.

### Tests written?
No

### Commits signed with GPG?
No

### CODE REVIEW

File: pkg/rpm/salt-minion [hunk 1]
```diff
@@ -116,8 +116,16 @@ start() {
     # $1 - config dir
     local retval=0
 
+    if [ -f $LOCK_FILE ]; then
+        echo "Service $SERVICE:$MINION_USER:$MINION_ID is starting"
```
Comment (cachedout) [inline pkg/rpm/salt-minion]: What happens if for some reason the minion crashes and the lockfile is not cleaned up? Won't the minion then fail to start correctly until it is manually removed?

Comment (rongzeng54) [reply pkg/rpm/salt-minion]: This lock file is different from the usual usage and it only exist within the startup process.
When the startup is completed, whether the result is successful or fails, the lock file will be cleaned up.

Comment (terminalmage) [reply pkg/rpm/salt-minion]: I don't like this either, it opens us up for problems later if the script is modified and the lockfile handling is not properly managed.

Comment (rongzeng54) [reply pkg/rpm/salt-minion]: hmm, seems in order to solve a small probability problem, a large probability problem is introduced. Before file lock management becomes reliable, i closed it.

Comment (rongzeng54) [reply pkg/rpm/salt-minion]: @terminalmage  @cachedout 
Now it seems that the problem can be solved.

**Comment by cachedout** [comment]
@dmurphy18 Do you have any thoughts here?

File: pkg/rpm/salt-minion (hunk 1 above)
Comment (dmurphy18) [reply pkg/rpm/salt-minion]: Sorry for delay in reviewing this, but the changes should be made in the salt-pack repository, see https://github.com/saltstack/salt-pack/blob/develop/file_roots/pkg/salt/2017_7/rhel6/sources/salt-minion for the 2017.7 branch and there is a similar change for the 2018.3 branch.

The contents of salt-minion are similar but different.  Need to update salt repository with a more recent version of packaging or pointers to salt-pack (salt-pack-py3 for Python3).

Further to the problem of cleanup on failure, perhaps should have a cleanup function added based off trapping signals, etc. which would ensure the lock file is removed.

**Review by dmurphy18 — CHANGES_REQUESTED** [review]
Sorry for delay in reviewing this, but the changes should be made in the salt-pack repository, see https://github.com/saltstack/salt-pack/blob/develop/file_roots/pkg/salt/2017_7/rhel6/sources/salt-minion for the 2017.7 branch and there is a similar change for the 2018.3 branch.

The contents of salt-minion are similar but different and any changes should be made in salt-pack.. 
SaltStack needs to update salt repository with a more recent version of packaging or pointers to salt-pack (salt-pack-py3 for Python3).

Further to the problem of cleanup on failure, perhaps there should be an additional cleanup function added based off trapping signals, etc. which would ensure the lock file is removed on exit.

**Comment by thatch45** [comment]
Given the status of this PR I am going to close it out. This change should be made to salt-pack and we should consider removing some of the pkg directory files wholesale

**Pull request closed without merging**